### PR TITLE
Overlays don't position themselves properly when target is an SVG

### LINKED ISSUE #1050: Overlays don't position themselves properly when target is an SVG
`getBoundingClientRect` should provide enough information to make it possible.

### PR DESCRIPTION
**Pull request by benekastah**
Fixes #1050

I started trying to get tests going for this, but it ended up taking a while for me for various reasons. I can come back to it later if needed.

### CODE REVIEW

File: src/utils/domUtils.js [hunk 1]
```diff
@@ -138,6 +138,24 @@ function getPosition(elem, offsetParent) {
 }
 
 /**
+ * Get an element's size
+ *
+ * @param {HTMLElement} elem
+ * @returns {{width: number, height: number}}
+ */
+function getSize(elem) {
+  if (typeof elem.getBoundingClientRect !== 'undefined') {
```
Comment (taion) [inline src/utils/domUtils.js]: This is not the right check. Non-compliant browsers may return an object that lacks the `width` and `height` entries.

**Comment by benekastah** [comment]
Not sure how to get the checks to pass. I ran tests locally, which all passed.

File: src/utils/domUtils.js (hunk 1 above)
Comment (benekastah) [inline src/utils/domUtils.js]: Ah, good point. I changed it. Does it look good to you now?

**Comment by taion** [comment]
Should make sure this plays well w/#1058, and unwind what's there if necessary.

**Comment by benekastah** [comment]
@taion Not sure if we still needed that change, but I kept it in `getSize` just in case.

**Comment by taion** [comment]
I'd drop it if your fix covers the Firefox + SVG use case. The less cruft the better.

**Comment by benekastah** [comment]
Ah, ok. I'll leave it as it was then.

**Comment by jquense** [comment]
do we need a test case for this? Seems like a nice thing to have, not sure what others feel like

**Comment by AlexKVal** [comment]
I would be nice to have tests for the new functionality:
`utils/domUtils.js`
```js
getSize(elem)
```

`utils/overlayPositionUtils.js`
```js
getContainerDimensions(containerNode)
getPosition(target, container)
```

**Comment by taion** [comment]
Well, we don't run CI against Firefox anyway, and this is really just a better implementation of existing behavior. Assuming the overlays still look right on the components docs page, I'm okay with this going in as is.

LGTM

**Comment by benekastah** [comment]
Ok, I merged the "Properly test position calculation" stuff in and got the tests working. Everything should be good to go.

File: src/utils/domUtils.js [hunk 2]
```diff
@@ -138,6 +138,25 @@ function getPosition(elem, offsetParent) {
...
+ * @returns {{width: number, height: number}}
+ */
+function getSize(elem) {
+  let rect = {
+    width: elem.offsetWidth,
+    height: elem.offsetHeight
+  };
+  if (typeof elem.getBoundingClientRect !== 'undefined') {
+    let {width, height} = elem.getBoundingClientRect();
+    rect.width = width || rect.width;
```
Comment (taion) [inline src/utils/domUtils.js]: What if gBCR actually returns `width` or `height` of 0?

Comment (benekastah) [inline src/utils/domUtils.js]: If that happens, `offsetWidth` or `offsetHeight` ought to be 0 too.

Comment (taion) [inline src/utils/domUtils.js]: Right, it's the case where `offsetWidth` is `null`-y rather than 0 that I'm concerned about here.

Comment (benekastah) [inline src/utils/domUtils.js]: Good call. Just fixed.

**Comment by taion** [comment]
LGTM.

Need one more reviewer from @react-bootstrap/collaborators.

**Comment by jquense** [comment]
LGTM

**Pull request merged**